src/main/java/io/pravega/connectors/flink/EventTimeOrderingFunction.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.connectors.flink;
+
+import org.apache.flink.api.common.state.MapState;
+import org.apache.flink.api.common.state.MapStateDescriptor;
+import org.apache.flink.api.common.state.ValueState;
+import org.apache.flink.api.common.state.ValueStateDescriptor;
+import org.apache.flink.api.common.typeinfo.BasicTypeInfo;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.java.typeutils.ListTypeInfo;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.streaming.api.functions.KeyedProcessFunction;
+import org.apache.flink.util.Collector;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class EventTimeOrderingFunction<T> extends KeyedProcessFunction<String, T, T> {
+
+    private static final long serialVersionUID = 1L;
+
+    private static final String EVENT_QUEUE_STATE_NAME = "eventQueue";
+
+    private static final String LAST_TRIGGERING_TS_STATE_NAME = "lastTriggeringTsState";
+
+    /**
+     * The input type information for buffering events to managed state.
+     */
+    private final TypeInformation<T> typeInformation;
+
+    /**
+     * The queue of input elements keyed by event timestamp.
+     */
+    private transient MapState<Long, List<T>> elementQueueState;
```

Comment:
It is a bit strange to see a map state with a comment saying it's a queue. It is actually a map contains all the data between watermarks. We can use the word dataState along with the name there.

Reply:
Change the name to `dataState` with comment `State to buffer all the data between watermarks.`